plugins/runner/internal/runtimehandler/binary/handler.go

```diff
@@ -0,0 +1,202 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+
+package binary
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+
+	"github.com/anchore/stereoscope"
+	multierror "github.com/hashicorp/go-multierror"
+	"github.com/openclarity/vmclarity/plugins/runner/internal/runtimehandler"
+	"github.com/openclarity/vmclarity/plugins/runner/types"
+	"github.com/openclarity/vmclarity/plugins/sdk-go/plugin"
+	"github.com/openclarity/vmclarity/scanner/utils/containerrootfs"
+)
+
+type binaryRuntimeHandler struct {
+	config types.PluginConfig
+
+	cmd       *exec.Cmd
+	cmdStdOut bytes.Buffer
+	cmdStdErr bytes.Buffer
+
+	pluginDir            string
+	inputDirMountPoint   string
+	pluginServerEndpoint string
+	ready                bool
+}
+
+func New(ctx context.Context, config types.PluginConfig) (runtimehandler.PluginRuntimeHandler, error) {
+	return &binaryRuntimeHandler{
+		config: config,
+	}, nil
+}
+
+func (h *binaryRuntimeHandler) Start(ctx context.Context) error {
+	h.pluginDir = filepath.Join(os.TempDir(), "vmclarity-plugins", h.config.Name)
+
+	image, err := stereoscope.GetImage(ctx, h.config.ImageName)
+	if err != nil {
+		return fmt.Errorf("unable to get image(%s): %w", h.config.ImageName, err)
+	}
+
+	if _, err := os.Stat(h.pluginDir); os.IsNotExist(err) {
+		err = containerrootfs.ToDirectory(ctx, h.config.ImageName, h.pluginDir)
+		if err != nil {
+			return fmt.Errorf("unable to extract image(%s): %w", h.config.ImageName, err)
+		}
+	}
+
+	// Mount input from host
+	// /tmp/vmclarity-plugins/kics + /input + /host-dir-to-scan
+	h.inputDirMountPoint = filepath.Join(h.pluginDir, runtimehandler.RemoteScanInputDirOverride, h.config.InputDir)
+	err = os.MkdirAll(h.inputDirMountPoint, 0755)
+	if err != nil {
+		return fmt.Errorf("unable to create directory for mount point: %w", err)
+	}
+
+	// nolint:typecheck
+	err = syscall.Mount(h.config.InputDir, h.inputDirMountPoint, "", syscall.MS_BIND, "")
+	if err != nil {
+		return fmt.Errorf("unable to mount input directory (%s - %s): %w", h.config.InputDir, h.inputDirMountPoint, err)
+	}
+
+	// https://lwn.net/Articles/281157/
+	// "the read-only attribute can only be added with a remount operation afterward"
+	err = syscall.Mount(h.config.InputDir, h.inputDirMountPoint, "", syscall.MS_BIND|syscall.MS_REMOUNT|syscall.MS_RDONLY, "")
+	if err != nil {
+		return fmt.Errorf("unable to remount input directory as read-only (%s - %s): %w", h.config.InputDir, h.inputDirMountPoint, err)
+	}
+
+	// TODO: handle panics - unmount
```

Comment:
If there is a panic, the mount will be stuck.

This can also happen with the docker binary handler (it won't stop the running plugin container on a panic).